src/app/outbound.rs

```diff
@@ -398,6 +398,223 @@ pub mod orig_proto_upgrade {
     }
 }
 
+pub mod require_identity_on_endpoint {
```

Comment:
nit: `on_endpoint` is redundant imo, since identity can only ever apply to endpoints.

Reply:
If it's just a nit I lean towards keep it, but I can remove it if you'd like. I still have #2543 in the back of my head, and am still under the impression this would add the option for identity on the source (or at least some way to represent both the local and peer identities in a connection).